rust/src/tests/posix_fallocate.rs

```diff
@@ -0,0 +1,83 @@
+use std::{fs::File, io::Write};
+
+use nix::{
+    errno::Errno,
+    fcntl::{open, posix_fallocate, OFlag},
+    sys::stat::{lstat, Mode},
+    unistd::Uid,
+};
+
+use crate::{
+    runner::context::FileType,
+    test::{FileSystemFeature, TestContext},
+    tests::{
+        assert_ctime_changed, assert_ctime_unchanged,
+        chmod::{chmod},
+    },
+};
+
+crate::test_case! {increase_empty, FileSystemFeature::PosixFallocate}
+fn increase_empty(ctx: &mut TestContext) {
+    let expected_size = 567;
+
+    let (path, file) = ctx.create_file_no_mode(OFlag::O_RDWR).unwrap();
+    posix_fallocate(file, 0, expected_size).unwrap();
+
+    let size = lstat(&path).unwrap().st_size;
+    assert_eq!(size, expected_size);
+}
+
+crate::test_case! {increase_non_empty, FileSystemFeature::PosixFallocate}
+fn increase_non_empty(ctx: &mut TestContext) {
+    let expected_offset = 20_000;
+    let expected_size = 3456;
```

Comment:
These two variables aren't really named correctly.  They're the arguments supplied to posix_fallocate, not what you expect to find with stat.  I suggest
```suggestion
    let expected_offset = 20_000;
    let expected_size = 3456;
```
```suggestion
    let offset = 20_000;
    let size = 3456;
```